app/models/timetable_slot.rb

```diff
@@ -14,4 +14,12 @@ def day=(new_day)
     super Day.to_numeric(new_day)
   end
 
+  def weeks
+    super.map {|pg_range| pg_range.to_range } if super
```

Comment:
1. Please use shortcuts:

```diff
-super.map {|pg_range| pg_range.to_range } if super
+super&.map(&:to_range)
```

- The first `&` is a [Safe navigation operator](https://ruby-doc.org/core-2.7.1/doc/syntax/calling_methods_rdoc.html#label-Safe+navigation+operator) (aka optional chaining operator).

- The second one, `&:`, is called a [Pretzel colon](https://technology.customink.com/blog/2015/06/08/ruby-pretzels/).

2. You’re returning a new array that is not linked with the persistable one, so changes of the array will not be persisted. Thus you should return an immutable array to make it clear. This can be done using `Object#freeze`.

```rb
super&.map(&:to_range).freeze
```

Reply:
These shortcuts are useful. Thanks!